cli/src/main.rs

```diff
@@ -602,18 +615,33 @@ fn test(skip_deploy: bool) -> Result<()> {
                 None
             }
         };
-
         let log_streams = stream_logs(&cfg.cluster.url())?;
 
+        let ts_config_exist = Path::new("tsconfig.json").exists();
+
         // Run the tests.
-        let exit = std::process::Command::new("mocha")
-            .arg("-t")
-            .arg("1000000")
-            .arg("tests/")
-            .env("ANCHOR_PROVIDER_URL", cfg.cluster.url())
-            .stdout(Stdio::inherit())
-            .stderr(Stdio::inherit())
-            .output()?;
+        let exit;
+        if ts_config_exist {
+            exit = std::process::Command::new("ts-mocha")
```

Comment:
We probably need to add `ts-mocha` to travis https://github.com/project-serum/anchor/blob/master/.travis.yml#L18